### PR TITLE
UpdateBuilding user documentation module: Update markdown to nh3

### PR DESCRIPTION
**Pull request by dpy013**
### Link to issue number:
Follow up to https://github.com/nvaccess/nvda/pull/16724
### Summary of the issue:
[markdown 3.5.1 to 3.6](https://github.com/Python-Markdown/markdown/releases/tag/3.6)
Updating python markdown to get the latest security updates
[modifications](https://github.com/messense/nh3/pull/35)
### Description of user facing changes
none
### Description of development approach
Updated markdown and nh3 versions of requirements.txt file
### Testing strategy:
Careful review of the generated documentation is required
### Known issues with pull request:
Not found at the moment
### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

## Summary by CodeRabbit

- **Chores**
  - Updated dependencies to improve security and stability:
    - Markdown updated from `3.5.1` to `3.6.0`.
    - nh3 updated from `0.2.15` to `0.2.17`.

### CODE REVIEW

**Comment by seanbudd** [comment]
this appears to be a duplicate of #16724 ?

**Comment by dpy013** [comment]
> this appears to be a duplicate of #16724 ?
PR 16724 has been abandoned.
As a new contributor, PR 16724 is not a reasonable pull request. It contains too many updates and lacks rigor.
As a result, I will be splitting pull request 16724 into smaller, more centralized parts and submitting only the useful and safe parts.
I apologize for the previous redundant pull request.

File: user_docs/en/changes.md [hunk 1]
```diff
@@ -37,6 +37,7 @@ Unicode CLDR has also been updated.
   * Updated LibLouis Braille translator to [3.30.0](https://github.com/liblouis/liblouis/releases/tag/v3.30.0). (#16652, @codeofdusk)
     * New Braille tables for Cyrillic Serbian, Yiddish, several ancient languages (Biblical Hebrew, Akkadian, Syriac, and Ugaritic), and transliterated Cuneiform text.
   * Updated NSIS 3.09 to 3.10 (#16674, @dpy013)
+  * Updated python markdown and nh3 to latest(#16725, @dpy013)
```
Comment (seanbudd) [inline user_docs/en/changes.md:40]: please add version numbers and new entries like the other items in this list

**Comment by seanbudd** [comment]
Can you please outline the testing you performed? What did the careful review involve?

**Comment by dpy013** [comment]
I did the following tests:
1.1 download [nvda_snapshot_pr16725-32533_changes.html](https://ci.appveyor.com/api/buildjobs/347mf09odad5xaj5/artifacts/output%2Fnvda_snapshot_pr16725-32533%2C44f5dd72_changes.html) and
[nvda_snapshot_pr16725-32559_changes.html](https://ci.appveyor.com/api/buildjobs/o8o8t32dxiiixv00/artifacts/output%2Fnvda_snapshot_pr16725-32559%2C24617178_changes.html)
1.2 Two files were opened using Microsoft Edge and chrome browsers.
The links, headings, and tables were tested to see if they could be viewed correctly with screen readers and browsers.
1.3 Finally tested [nvda pr16725](https://ci.appveyor.com/api/buildjobs/o8o8t32dxiiixv00/artifacts/output%2FkeyCommands.html)
and
[userGuide.html](https://ci.appveyor.com/api/buildjobs/o8o8t32dxiiixv00/artifacts/output%2FuserGuide.html)
The links, tables, and forms can be viewed correctly with screen readers and browsers.

**Comment by seanbudd** [comment]
Thanks @dpy013

**Pull request merged**